vhdbuilder/packer/trivy-scan.sh

```diff
@@ -1,41 +1,142 @@
 #!/usr/bin/env bash
 set -euxo pipefail
 
-TRIVY_REPORT_JSON_PATH=/opt/azure/containers/trivy-report.json
-TRIVY_REPORT_TABLE_PATH=/opt/azure/containers/trivy-images-table.txt
+TRIVY_REPORT_DIRNAME=/opt/azure/containers
+TRIVY_REPORT_ROOTFS_JSON_PATH=${TRIVY_REPORT_DIRNAME}/trivy-report-rootfs.json
+TRIVY_REPORT_IMAGE_TABLE_PATH=${TRIVY_REPORT_DIRNAME}/trivy-report-images-table.txt
+
 TRIVY_VERSION="0.40.0"
 TRIVY_ARCH=""
 
+OS_SKU=${1}
+OS_VERSION=${2}
+TEST_VM_ADMIN_USERNAME=${3}
+ARCHITECTURE=${4}
+SIG_CONTAINER_NAME=${5}
+STORAGE_ACCOUNT_NAME=${6}
+ENABLE_TRUSTED_LAUNCH=${7}
+VHD_NAME=${8}
+SKU_NAME=${9}
+KUSTO_ENDPOINT=${10}
+KUSTO_DATABASE=${11}
+KUSTO_TABLE=${12}
+TRIVY_UPLOAD_REPORT_NAME=${13}
+TRIVY_UPLOAD_TABLE_NAME=${14}
+
+if [[ "$OS_SKU" == "Ubuntu" ]] && [[ "$OS_VERSION" == "20.04" ]]; then
+    sudo apt-get install -y azure-cli
+elif [[ "$OS_SKU" == "Ubuntu" ]] && [[ "$OS_VERSION" == "22.04" ]] && [[ "${ARCHITECTURE,,}" == "arm64" ]]; then
+    sudo apt update
+    sudo apt install -y python3-pip
+    pip install azure-cli
+    export PATH="/home/$TEST_VM_ADMIN_USERNAME/.local/bin:$PATH"
+    CHECKAZ=$(pip freeze | grep "azure-cli==")
+    if [[ -z $CHECKAZ ]]; then
+        echo "Azure CLI is not installed properly."
+        exit 1
+    fi
+elif [[ "$OS_SKU" == "Ubuntu" ]] && [[ "$OS_VERSION" == "22.04" ]] && [[ "${ARCHITECTURE,,}" != "arm64" ]]; then
```

Comment:
I know @AlisonB319 is working on onboarding 2404 to scanning, might want to sync with her on this

Reply:
yep, thanks for the reminder